dateparser/freshness_date_parser.py

```diff
@@ -13,7 +16,40 @@ class FreshnessDateDataParser(object):
     def __init__(self, now=None):
         self.now = now or datetime.utcnow()
 
+    def _are_all_words_units(self, date_string):
+        skip = [_UNITS,
+                r'about|ago|\d+',
+                r':|[ap]m'] 
+
+        date_string = re.sub(r'\s+', ' ', date_string.strip())
+
+        words = filter(lambda x: x if x else False, re.split('\W', date_string))
+        words = filter(lambda x: not re.match(r'%s' % '|'.join(skip), x), words)
+        return not bool(words)
+
+    def _parse_time(self, date_string):
+        date_string = re.sub(r'\d+\s*(%s)' % _UNITS, '', date_string)
+        date_string = re.sub(r'[\.,]', '', date_string)
+        try:
+            return parse(date_string).time()
+        except:
+            pass
+
     def parse(self, date_string):
+        time = self._parse_time(date_string)
```

Comment:
I supposed are_all_words_units would exit the date parser if no parseable information is found. Sure let me still put it inside `if date:`.